examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -66,6 +51,14 @@ CHIP_ERROR PairingCommand::RunInternal(NodeId remoteId)
 
     InitCallbacks();
     UpdateWaitForResponse(true);
+
+    //
+    // We're about to call methods into the stack, so lock
+    // appropriately. None of the following calls below before the unlock are, nor should be
+    // blocking.
+    //
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
```

Comment:
Suggest wrapping the locking and the switch into a new scope block just to make it clearer and also simplify RAII locking later.

Reply:
As part of RAII-ification in this PR, I'll push this into a scoped block.